### PR TITLE
Added require* extensions

### PR DESCRIPTION
**Pull request by Miha-x64**
When developing for a specific device, you typically know its protocol, i. e. UUIDs of services, characteristics, and descriptors. Thus you may write something like `gatt.getService(sUuid)!!.getCharacteristic(cUuid)!!`.
This PR adds `require*()` extensions to replace such code with `gatt.requireService(sUuid).requireCharacteristic(cUuid)`, or even `gatt.requireCharacteristic(sUuid, cUuid)`.

### CODE REVIEW

**Comment by LouisCAD** [comment]
Thanks for the PR! I think it could use `requireNotNull` from the stdlib. If it works, it could be great to use it.
That's a great idea anyway, and could actually be useful for most projects.

I'll review this pull request thoroughly in several days.

File: core/src/main/java/com/beepiz/bluetooth/gattcoroutines/experimental/extensions/GattService.kt [hunk 1]
```diff
@@ -4,9 +4,49 @@ import android.bluetooth.BluetoothGattService
 import android.os.Build.VERSION_CODES.JELLY_BEAN_MR2
 import android.support.annotation.RequiresApi
 import com.beepiz.bluetooth.gattcoroutines.experimental.BGC
+import com.beepiz.bluetooth.gattcoroutines.experimental.BGD
+import com.beepiz.bluetooth.gattcoroutines.experimental.GattConnection
 import java.util.*
```
Comment (twyatt) [inline core/src/main/java/com/beepiz/bluetooth/gattcoroutines/experimental/extensions/GattService.kt:9]: @LouisCAD Although not part of this PR, it is generally advised against using `*` for imports.

> Wildcard imports (of any type) are not allowed.

https://developer.android.com/kotlin/style-guide

Perhaps worth changing outside of this PR?

Comment (LouisCAD) [reply core/src/main/java/com/beepiz/bluetooth/gattcoroutines/experimental/extensions/GattService.kt:9]: @twyatt This implies changing the IDE to not auto use wildcard import for this specific `java.util` package, but is really out of the scope of this PR. Feel free to create an issue for this if you want.

**Comment by LouisCAD** [comment]
Merged. Thanks! Will be released on jcenter soon.
